src/openrct2/rct1/S4Importer.cpp

```diff
@@ -829,11 +829,11 @@ namespace RCT1
             dst->num_stations = src->num_stations;
 
             // Vehicle links (indexes converted later)
-            for (int32_t i = 0; i < RCT1_MAX_TRAINS_PER_RIDE; i++)
+            for (int32_t i = 0; i < Limits::MaxTrainsPerRide; i++)
             {
                 dst->vehicles[i] = src->vehicles[i];
             }
-            for (int32_t i = RCT1_MAX_TRAINS_PER_RIDE; i <= MAX_VEHICLES_PER_RIDE; i++)
+            for (int32_t i = Limits::MaxTrainsPerRide; i < OpenRCT2::Limits::MaxTrainsPerRide; i++)
```

Comment:
Don’t change the `<=`, it is there for a reason - to properly mark all vehicles as NULL.

If you absolutely must change this line, use std::size instead.

Reply:
The change is correct. See my comments in ride.h